src/sGLOH2.cpp

```diff
@@ -7,157 +7,190 @@
+        // Extract the patch around the keypoint.
+        // The patch is a square region of the image with side length patchSize and top-left corner at topLeft.
         cv::Mat patch = image(cv::Rect(topLeft.x, topLeft.y, patchSize, patchSize));
 
-        // Compute sGLOH descriptor for the patch
+        // Compute sGLOH descriptor for the patch.
+        // The sGLOH descriptor is a modification of the GLOH descriptor that is more robust to changes in rotation.
         cv::Mat descriptor = compute_sGLOH(patch);
 
-        // Append descriptor to descriptors matrix
+        // Append descriptor to descriptors matrix.
+        // The descriptors matrix is a matrix where each row is a descriptor of a keypoint.
         descriptors.push_back(descriptor);
     }
 }
 
 cv::Mat sGLOH2::compute_sGLOH_single(const cv::Mat& patch) {
+    // Initialize the descriptor as a single-row matrix of zeros.
+    // The type of the matrix is CV_32F, which means that it contains 32-bit floating-point numbers.
     cv::Mat descriptor = cv::Mat::zeros(1, 0, CV_32F);
 
-    // Compute the size of the regions
+    // Compute the size of the regions into which the patch will be divided.
+    // Each region will be a square with side length equal to the number of rows in the patch divided by m.
     int region_size = patch.rows / m;
 
-    // Define the bin edges for the histogram
-    std::vector<float> binEdges = {0, 90, 180, 270, 360};  // Fill this with the desired bin edges
-
-    // Compute the histogram for each region
+    // Compute the histogram for each region.
+    // The patch is divided into m*m regions, and a histogram is computed for each one.
     for (int r = 0; r < m; r++) {
         for (int d = 0; d < m; d++) {
-            // Compute the region
+            // Compute the region.
+            // The region is a square with top-left corner at (r*region_size, d*region_size) and side length region_size.
             cv::Rect region(r * region_size, d * region_size, region_size, region_size);
 
-            // Compute the histogram for the region
-            cv::Mat histogramMat = computeCustomHistogram(patch(region), binEdges);
+            // Compute the histogram for the region.
+            // The computeHistogram function computes a histogram of gradient orientations for the region.
+            cv::Mat histogramMat = computeHistogram(patch(region), m);
 
             // Concatenate the histogram to the descriptor.
+            // The hconcat function horizontally concatenates the histogram to the descriptor.
+            // This means that the histogram is added as new columns at the end of the descriptor.
             cv::hconcat(descriptor, histogramMat, descriptor);
         }
     }
 
+    // Return the final descriptor.
+    // The descriptor is a single-row matrix that contains the concatenated histograms of all regions in the patch.
     return descriptor;
 }
 
+
 cv::Mat sGLOH2::compute_sGLOH(const cv::Mat& patch) {
```

Comment:
Might be worth clarifying that this computes the enhanced sGLOH or sGLOH 2 with the rotation